cmd/juju/application/addrelation.go

```diff
@@ -34,28 +44,26 @@ func (c *addRelationCommand) Info() *cmd.Info {
 }
 
 func (c *addRelationCommand) Init(args []string) error {
+	// Must have only 2 arguments
 	if len(args) != 2 {
 		return errors.Errorf("a relation must involve two applications")
 	}
+	// None of the arguments can be empty.
+	if args[0] == "" || args[1] == "" {
+		return errors.Errorf("a relation must involve two applications")
```

Comment:
Considering that this PR is about improving tests, I'll remove the validation that I have added for command arguments. All validation of application/endpoints happens at state as there are several permutations to consider and I do not want to hold up this PR from landing.

I'll adjust tests accordingly - just check that the correct number of arguments passed into the command.